### DISCUSSION TITLE
[FEATURE] Layout Engines

### DISCUSSION THREAD

**Discussion by amcdnl** [xyflow/xyflow#4252] [Layouting]
## Request
It would be cool if you could support both custom drag and layout engines. 

I would suggest looking at:

- ELKJS ( my fav ) https://www.npmjs.com/package/elkjs
- Cola - https://github.com/tgdwyer/WebCola
- Dagre - https://github.com/dagrejs/dagre

**Comment by moklick** [comment]
hey @amcdnl,

thanks for your input :) We already have an issue regarding layouting #5 .
Could you open a new issue for the custom drag engine feature request?

**Comment by amcdnl** [comment]
Sure! A side note on the ElkJS - its the best layout for this kind of stuff but the api is AWFUL to work with.
